### PR TITLE
[eas-json] improve errors and error message formatting

### PR DESCRIPTION
**Pull request by Simek**
# Checklist

- [ ] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

This small PR focuses on improving custom errors emitted by `eas-json`, the main goal was to improve the readability.

# How

Set the error name to the constructor name, decorate base error and few of the messages using `chalk`.

# Test Plan

The changes have been tested by running locally build `eas`.

# Preview

Screenshot 2022-09-29 003303

### CODE REVIEW

File: packages/eas-json/src/accessor.ts [hunk 1]
```diff
@@ -90,7 +96,7 @@ export class EasJsonAccessor {
       this.isJson5 = true;
       return rawJSON;
     } catch (originalError: any) {
-      const err = new Error('Found invalid JSON in eas.json.');
+      const err = new InvalidEasJsonError(`Found invalid character in ${chalk.bold('eas.json')}.`);
```
Comment (wkozyra95) [inline packages/eas-json/src/accessor.ts:99]: Are you sure this message will always make sense? e.g. what happens if a file is empty?

**Review by wkozyra95 — APPROVED** [review]
fix tests

File: packages/eas-json/src/accessor.ts (hunk 1 above)
Comment (Simek) [reply packages/eas-json/src/accessor.ts:99]: Screenshot 2022-09-29 125153

Maybe we can separate this edge case, I will look at it.

**Comment by Simek** [comment]
/changelog-entry chore Improve errors and error messages formatting related to **eas.json**.

**Review by wkozyra95 — APPROVED** [review]

File: packages/eas-json/src/accessor.ts (hunk 1 above)
Comment (Simek) [reply packages/eas-json/src/accessor.ts]: Finally I have changed the logic a bit and reverted the original error message, when there is no information about problem location in the file.
